### ISSUE TITLE
Storage: Improve local storage error handling

### ISSUE THREAD

**Issue by ifrost** [grafana/grafana#40434]
**What happened**:

In numerous places local storage is updated without error handling. When quota is exceeded it may cause the app to break (like [here](https://github.com/grafana/grafana/issues/38814)) or if the error is caught by a component at some level it may break the component like below:

![image](https://user-images.githubusercontent.com/745532/137269155-37faf1c3-5976-4cbe-aafe-5a4ffefb8312.png)

What is more, Grafana is inaccessible when local storage is not available:

- https://github.com/grafana/grafana/issues/33610
- https://github.com/grafana/grafana/issues/15884

**What you expected to happen**:

If local storage is not available or full:
- Grafana should not break. 
- Values should be persisted only for the duration of the current session.
- Maybe the user should be informed about the error?

**How to reproduce it (as minimally and precisely as possible)**:

- Add artificial items to the storage to increase taken space and try to save new items.
- Disable local storage and access Grafana (e.g. running `/Applications/Google\ Chrome.app/Contents/MacOS/Google\ Chrome --disable-local-storage`)

**Anything else we need to know?**:

**Environment**:
- Grafana version: 8.2.x
- Data source type & version: any
- OS Grafana is installed on: any
- User OS & Browser: any
- Grafana plugins: N/A
- Others: N/A

**Comment by joshhunt** [comment]
We do have the `app/core/store.ts` which could serve as the interface for everything using local storage goes through, with better fallback for when localStorage is unavailable, but unfortuantely not everyone uses this. 

Would be great to add better fallback to that class, and enforce it's usage through linting.

**Comment by Clarity-89** [comment]
@tskarhed has this been fixed by the localStorage work you did?

**Comment by tskarhed** [comment]
Yes, I would say this is fixed. We've prevented further new errors of this kind at the very least.
